### PR TITLE
Add tf.function(jit_compile=True) for the PNG decode function.

### PR DESCRIPTION
**Pull request by qlzh727**
# What does this PR do?
Change the decode function to be backed with XLA for better performance.

See https://github.com/keras-team/keras-cv/pull/943#discussion_r1001092882 for the actual performance result. 

In short, XLA compile introduce 0.4 sec overhead, but save 0.2 sec per 1000 examples.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

### CODE REVIEW

**Comment by qlzh727** [comment]
@bhack

**Review by bhack — APPROVED** [review]
Just a side note. I don't know if you want to comment inline...
Do you think that this function could work in batch in the case it will be transformed in a KPL?
This just eventually in the case we have a CPU bottleneck and we starve GPU cycles for input images.

File: keras_cv/datasets/pascal_voc/segmentation.py [hunk 1]
```diff
@@ -259,6 +259,7 @@ def _build_metadata(data_dir, image_ids):
     return result
 
 
+@tf.function(jit_compile=True)
```
Comment (tanzhenyu) [inline keras_cv/datasets/pascal_voc/segmentation.py:262]: Maybe add a comment here for some data: w/ or w/o the change, what happens, i.e., compile time change, process time change, etc (though it's highly hardware dependent)

**Review by tanzhenyu — APPROVED** [review]

File: keras_cv/datasets/pascal_voc/segmentation.py (hunk 1 above)
Comment (bhack) [reply keras_cv/datasets/pascal_voc/segmentation.py:262]: Yes in theory we have git blame to got to this PR and then to the benchmark comment but not many users has this habit... :smiley_cat:

Comment (qlzh727) [reply keras_cv/datasets/pascal_voc/segmentation.py:265]: Done. Added comment.

**Comment by qlzh727** [comment]
> Just a side note. I don't know if you want to comment inline... Do you think that this function could work in batch in the case it will be transformed in a KPL? This just eventually in the case we have a CPU bottleneck and we starve GPU cycles for input images.

Are you referring to tf.vectorized_map in the KPL? This function is not used in the KPL, and I think the batched dataset doesn't use tf.vectorized_map as implementation.

**Comment by bhack** [comment]
> Are you referring to tf.vectorized_map in the KPL? This function is not used in the KPL,

No I meant these colormapping inversions could be exactly KPLs, like a crop or any other preprocessing layer we have.
Than we could rely on KPL tf.vectorized_map, in this case I think it will work without any fallback, but probably this function it is quite "batch ready" to work with batches directly.

**Comment by bhack** [comment]
Other then this KPL hypothesis (for the GPU) also considering to be still in `tf.dataset` I think that if it is easy to have a batched version of that function we could then batch before map `_decode_png_mask` instead of only parallel mapping the whole load:

// dataset = dataset.map(_load_images, num_parallel_calls=tf.data.AUTOTUNE) 

See https://www.tensorflow.org/guide/data_performance#vectorizing_mapping

**Comment by bhack** [comment]
Just to make a small gist example if you want still maintain this in the dataset processing (in the case you don't want to transform this in a KPL to run it on GPU):

https://colab.research.google.com/gist/bhack/2afa97dcb7455ee3edd23ffe29ec93e9/vectorized_colormappign_inversion.ipynb

Of course you can still maintain at the same time `dataset.map(_load_images, num_parallel_calls=tf.data.AUTOTUNE)` that I've just removed to simplify the example.

**Comment by qlzh727** [comment]
Thanks for the example. 

I think the current function can be executed correctly with batched pngs. At this stage, we will keep the dataset as is so that user can add batch/shuffle/cache by themselves down the stream. If we choose to extend the `load()` API to include more params for batch/shuffle or more, we can certainly move this function after batch and let it process the batched pngs.

Eventually, I think we will provide a function to convert the current dataset into tfrecords format, and user will just load the converted data from it, without further decoding, etc.

**Comment by bhack** [comment]
> I think the current function can be executed correctly with batched pngs

Yes the function it is already running in batches "slightly modified" (as you don't have the diff) in the colab
